topics/client.md

```diff
@@ -1,105 +1,204 @@
 
-For example, you can add a `CIO` engine dependency as follows:
+### Engine Dependency {id="engine-dependency"}
+An engine is in charge of processing network requests. There are different client engines available for [various platforms](http-client_multiplatform.md), such as Apache, CIO, Android, iOS, and so on. For example, you can add a `CIO` engine dependency as follows:
 <var name="artifact_name" value="ktor-client-cio"/>
 <include src="lib.md" include-id="add_ktor_artifact"/>
 
-## Creating client
+For a full list of dependencies required for a specific engine, see [](http-client_engines.md#dependencies).
+
+### Feature Dependency {id="feature-dependency"}
+If you want to use additional client features, you need to add a corresponding dependency. You can learn which dependencies you need from a topic for a required feature.
+
 
-Next you can create a client as here:
+
+
+## Create the Client {id="create-client"}
+
+To instantiate the client, create the [HttpClient](https://api.ktor.io/%ktor_version%/io.ktor.client/-http-client/index.html) class instance and pass an engine as a parameter:
 
 ```kotlin
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+
 val client = HttpClient(CIO)
 ```
 
-where `CIO` is engine class here. If you confused which engine class you should use consider using `CIO`.
-
-If you're using multiplatform, you can omit the engine:
+In this example, we use the [CIO](https://api.ktor.io/%ktor_version%/io.ktor.client.engine.cio/-c-i-o/index.html) engine. You can also omit an engine:
 
 ```kotlin
+import io.ktor.client.*
+
 val client = HttpClient()
 ```
 
-Ktor will choose an engine among the ones that are available from the included artifacts using a `ServiceLoader` on the JVM, or similar approach in the other platforms. If there are multiple engines in the dependencies Ktor chooses first in alphabetical order of engine name.
+In this case, the client will choose an engine automatically depending on the artifacts [added in a build script](#engine-dependency). You can learn more from the [Engines](http-client_engines.md#default) topic.
 
-It's safe to create multiple instance of client or use the same client for multiple requests.
+## Configure the Client {id="configure-client"}
 
-## Releasing resources
+### Basic Configuration {id="basic-config"}
 
-Ktor client is holding resources: prepared threads, coroutines and connections. After you finish working with the client, you may wish to release it by calling `close`:
+To configure the client, you can pass an additional functional parameter to the client constructor. The [HttpClientEngineConfig](https://api.ktor.io/%ktor_version%/io.ktor.client.engine/-http-client-engine-config/index.html) class is a base class for configuring the client. In the example below, receiving HTTP errors in response don't cause exceptions:
```

Comment:
This seems a bit strange. "In the exmaple below,..."

maybe indicate "For instance, we can change the behaviour of exceptions in the Basic Configuration:"